### PR TITLE
Add Docker support

### PR DESCRIPTION
**Pull request by rickycraft**
I thought docker could be useful.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,6 +3,10 @@
   "version": "1.0.0",
   "description": "Servizio che consente a tutti gli studenti Unibo di aggiungere l'orario delle loro lezioni, filtrando solo quelle di interesse, al proprio calendario. Dopo aver aggiunto il link del proprio orario e l'anno di corso, il servizio permette di selezionare solo i corsi da frequentare e genera un semplice link con il quale eseguire una \"Iscrizione a Calendario\" nel proprio calendario personale.",
   "main": "index.js",
+  "scripts": {
+    "start": "node index.js",
+    "build": "sudo docker build -t unibocalendar:latest ."
```
Comment (eutampieri) [inline package.json:8]: Please don't use sudo since `usermod -aG docker` is a much nicer way

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+services:
+  unibocalendar:
+    container_name: "unibocalendar"
+    image: "unibocalendar"
+    restart: unless-stopped
+    ports:
+      - "3000:3000"
+    volumes:
+      - "./db:/app/logs"
+      - "./data:/app/opendata"
```
Comment (eutampieri) [inline docker-compose.yml:12]: Good, I assume this is to ease the configuration since you already configured the right volumes.

File: index.js [hunk 3]
```diff
@@ -55,10 +53,9 @@ db.run("CREATE TABLE IF NOT EXISTS requested_lectures (enrollment_id TEXT, lectu
 db.run("CREATE TABLE IF NOT EXISTS hits (date INTEGER, enrollment_id TEXT, user_agent TEXT)");
 db.run("CREATE TABLE IF NOT EXISTS token(id TEXT, description TEXT)");
 db.run("CREATE TABLE IF NOT EXISTS cache(id TEXT, value TEXT, expiration INTEGER)");
-//csvMigrator.migrate(db);
 db.close()
 
 //start server
-app.listen(app.get("port"), "127.0.0.1", function () {
-    console.log(`UniboClendar started on http://127.0.0.1:${app.get("port")}; press Ctrl-C to terminate.`);
+app.listen(app.get("port"), "0.0.0.0", () => {
```
Comment (eutampieri) [inline index.js:59]: Actually, an environment variable with localhost as a fallback would be better than defaulting to ::/0

File: package.json [hunk 4]
```diff
@@ -25,4 +29,4 @@
   },
   "homepage": "https://github.com/FrancescoBonzi/UniboCalendar#readme",
   "type": "module"
-}
+}
```
Comment (eutampieri) [inline package.json:32]: Add newline at the end of file

File: index.js [hunk 5]
```diff
@@ -55,10 +53,9 @@ db.run("CREATE TABLE IF NOT EXISTS requested_lectures (enrollment_id TEXT, lectu
...
+    console.log(`UniboClendar started on http://0.0.0.0:${app.get("port")}`);
 });
```
Comment (eutampieri) [inline index.js:61]: We like newlines at the end of files

File: .dockerignore [hunk 6]
```diff
@@ -0,0 +1,3 @@
+node_modules
+opendata
+logs
```
Comment (eutampieri) [inline .dockerignore:3]: Newline

**Review by eutampieri — CHANGES_REQUESTED** [review]
Good job, maybe resolve the conversations. Waiting for @FrancescoBonzi 's feedback too.

File: index.js [hunk 7]
```diff
@@ -27,6 +27,9 @@ function tokenMiddleware(correct) {
 //set up port
 app.set("port", process.env.PORT || 3002);
 
+//set up listen ip
+app.set("listen-ip", process.env.LISTEN_IP || "localhost");
```
Comment (rickycraft) [inline index.js:31]: If you have a suggestion for a better variable name, i'm up to it

File: docker-compose.yml (hunk 2 above)
Comment (rickycraft) [reply docker-compose.yml:12]: Maybe adding a mention on the README could be useful. Also maybe providing also a plain docker command.

File: Dockerfile [hunk 8]
```diff
@@ -1,4 +1,4 @@
-FROM node:16
+FROM node:18-alpine
```
Comment (rickycraft) [inline Dockerfile:1]: Alpine image is lighter and works the same from my testing

Comment (rickycraft) [reply Dockerfile:1]: node 18 is currently LTS and same as before it works fine

**Comment by eutampieri** [comment]
ok
Il giorno 05/nov/22, alle ore 15:42, Riccardo Roveri ha scritto:

**Comment by eutampieri** [comment]
I wholly agree and approove. Alpine FTW!

**Comment by eutampieri** [comment]
Maybe BIND_ADDR?
Il giorno 05/nov/22, alle ore 15:38, Riccardo Roveri ha scritto:

File: index.js (hunk 7 above)
Comment (eutampieri) [reply index.js:31]: BIND_ADDR

**Review by eutampieri — CHANGES_REQUESTED** [review]
Almost there ;)

**Comment by rickycraft** [comment]
Do you want to update the README with docker infos?

**Comment by eutampieri** [comment]
> Do you want to update the README with docker infos?

This is up to @FrancescoBonzi

**Review by eutampieri — APPROVED** [review]

File: .dockerignore [hunk 9]
```diff
@@ -0,0 +1,3 @@
+node_modules
+opendata
```
Comment (eutampieri) [inline .dockerignore:2]: Add .git

**Review by eutampieri — CHANGES_REQUESTED** [review]

**Comment by FrancescoBonzi** [comment]
Grazie mille @rickycraft per il lavoro!! Super apprezzato 😊

**Pull request merged**